### DISCUSSION TITLE
Allow Individual Row Height Customisation in Grid Layout

### DISCUSSION THREAD

**Discussion by pbower** [tailwindlabs/tailwindcss#14098] [Ideas]
**Description**:

Currently, it seems difficult to customise the height of individual rows in a grid layout. The documentation doesn't provide examples for achieving this, and the current row count limitation makes it challenging to implement certain layouts.

**Use Case:**

I need to set my rows to be 3-4 times the height of the header row. However, it appears impossible to set the first (header) row to be smaller while maintaining the relative sizing needed for subsequent rows. For example, if the header row is set smaller, I can only fit three rows after it (due to Tailwind supporting 12 rows?), whereas I need at least five that are three times as tall as the header.

**Current Limitation:**

The documentation suggests making one row take a single unit and others take multiple units for relative sizing, but this quickly exhausts the available row units. It seems like such a basic requirement, so I might be missing something if this is a world-class library.

**Request:**

Please clarify if this functionality is currently available, and if not, consider adding support for individual row height customisation within the grid layout. Note that using Flexbox is not a suitable workaround, as the grid layout should inherently support this functionality.

Thank you!

**Comment by wongjn** [comment]
The `grid-rows-*` utility maps to the [`grid-template-rows`](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-rows) CSS property, so you can [use your own CSS value](https://tailwindcss.com/docs/grid-template-rows#using-custom-values) to have the row sizing that you'd like.
